nrf_security/src/mbedtls/CMakeLists.txt

```diff
@@ -1,20 +1,40 @@
 #
-# Copyright (c) 2019 Nordic Semiconductor
-#
 # SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
 #
-FUNCTION(PREPEND var prefix)
-  SET(listVar "")
+
+#
+# Internal function to append all unnamed parameters with 'prefix'
+# and add it to the list 'var'
+#
+FUNCTION(append_with_prefix var prefix)
+  SET(listVar ${${var}})
   FOREACH(f ${ARGN})
+    kconfig_mbedtls_debug("Adding: " ${f} " to " ${var})
     LIST(APPEND listVar "${prefix}${f}")
   ENDFOREACH(f)
-  SET(${var} "${listVar}" PARENT_SCOPE)
-ENDFUNCTION(PREPEND)
+  SET(${var} ${listVar} PARENT_SCOPE)
+ENDFUNCTION(append_with_prefix)
 
-# ----------  Original (unmodified) mbedTLS sources -------------
+#
+# Internal function to append all unanamed parameters with 'prefix'
+# and add it to the list 'var' if 'condition' is met
+#
+FUNCTION(append_with_prefix_ifdef condition var prefix)
+  IF(${condition})
+    kconfig_mbedtls_debug("Condition: " ${condition} " satisfied")
+    SET(listVar ${${var}})
+    FOREACH(f ${ARGN})
+      kconfig_mbedtls_debug("Adding: " ${f} " to " ${var})
+      LIST(APPEND listVar "${prefix}${f}")
+    ENDFOREACH(f)
+    SET(${var} ${listVar} PARENT_SCOPE)
+  ENDIF()
```

Comment:
Can append_with_prefix_ifdef call append_with_prefix ?

Reply:
Tried that, apparently this leads to scoping issues, but i can rewrite the logic fo this a bit together with lower-casing the keywords